src/wallet/wallet.cpp

```diff
@@ -2756,7 +2756,7 @@ bool CWallet::CreateTransaction(interfaces::Chain::Lock& locked_chain, const std
                 //  post-backup change.
 
                 // Reserve a new key pair from key pool
-                if (IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
+                if (!CanGetAddresses(true)) {
                     strFailReason = _("Can't generate a change-address key. Private keys are disabled for this wallet.");
```

Comment:
Nit: drop / tweak "Private keys are disabled for this wallet."

Reply:
Changed.